front/lib/api/assistant/feedback.ts

```diff
@@ -26,164 +23,118 @@ export type AgentMessageFeedbackType = {
   userId: number;
   thumbDirection: AgentMessageFeedbackDirection;
   content: string | null;
+  createdAt: Date;
+  agentConfigurationId: string;
+  agentConfigurationVersion: number;
+  isConversationShared: boolean;
+};
+
+export type AgentMessageFeedbackWithMetadataType = AgentMessageFeedbackType & {
+  conversationId: string | null;
+  userName: string;
+  userEmail: string;
+  userImageUrl: string | null;
 };
 
 export async function getConversationFeedbacksForUser(
   auth: Authenticator,
   conversation: ConversationType | ConversationWithoutContentType
-): Promise<Result<AgentMessageFeedbackType[], ConversationError>> {
+): Promise<Result<AgentMessageFeedbackType[], ConversationError | Error>> {
   const owner = auth.workspace();
   if (!owner) {
-    throw new Error("Unexpected `auth` without `workspace`.");
+    return new Err(new Error("workspace_not_found"));
   }
   const user = auth.user();
   if (!canAccessConversation(auth, conversation) || !user) {
     return new Err(new ConversationError("conversation_access_restricted"));
   }
 
-  const messages = await Message.findAll({
-    where: {
-      conversationId: conversation.id,
-      agentMessageId: {
-        [Op.ne]: null,
-      },
-    },
-    attributes: ["sId", "agentMessageId"],
-  });
-
-  const agentMessages = await AgentMessage.findAll({
-    where: {
-      id: {
-        [Op.in]: messages
-          .map((m) => m.agentMessageId)
-          .filter((id): id is number => id !== null),
-      },
-    },
-  });
-
   const feedbacks =
-    await AgentMessageFeedbackResource.fetchByUserAndAgentMessages(
-      user,
-      agentMessages
+    await AgentMessageFeedbackResource.getConversationFeedbacksForUser(
+      auth,
+      conversation
     );
 
-  const feedbacksByMessageId = feedbacks.map(
-    (feedback) =>
-      ({
-        id: feedback.id,
-        messageId: messages.find(
-          (m) => m.agentMessageId === feedback.agentMessageId
-        )!.sId,
-        agentMessageId: feedback.agentMessageId,
-        userId: feedback.userId,
-        thumbDirection: feedback.thumbDirection,
-        content: feedback.content,
-      }) as AgentMessageFeedbackType
-  );
-
-  return new Ok(feedbacksByMessageId);
+  return feedbacks;
 }
 
 /**
  * We create a feedback for a single message.
  * As user can be null (user from Slack), we also store the user context, as we do for messages.
  */
-export async function createOrUpdateMessageFeedback(
+export async function upsertMessageFeedback(
   auth: Authenticator,
   {
     messageId,
     conversation,
     user,
     thumbDirection,
     content,
+    isConversationShared,
```

Comment:
I don't see where it's used a no caller set it. What is the purpose of this?

Reply:
Is this supposed to be the boolean that tracks the confirmation by the user that they are ok sharing their conversation?

Reply:
Yes it is, it will be used later when I introduce further changes to Sparkle and front